### PR TITLE
Improve Documentation for `PreferenceContribution` and related types

### PR DESCRIPTION
**Pull request by tortmayr**
Signed-off-by: Tobias Ortmayr <tortmayr@eclipsesource.com>
Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 1]
```diff
@@ -31,6 +31,31 @@ import { Mutable } from '../../common/types';
 /* eslint-disable guard-for-in, @typescript-eslint/no-explicit-any */
 
 export const PreferenceContribution = Symbol('PreferenceContribution');
+
+/**
+ * A {@link PreferenceContribution} can be used to add additional custom preferences.
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:36]:
```suggestion
 * A {@link PreferenceContribution} allows adding additional custom preferences.
```

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 2]
```diff
@@ -31,6 +31,31 @@ import { Mutable } from '../../common/types';
...
+ * For this, the {@link PreferenceContribution} has to provide a valid JSON Schema which describes which preferences
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:37]:
```suggestion
 * For this, the {@link PreferenceContribution} has to provide a valid JSON Schema specifying which preferences
```

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 3]
```diff
@@ -31,6 +31,31 @@ import { Mutable } from '../../common/types';
...
+ * are available including types and description.
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:38]:
```suggestion
 * are available including their types and description.
```

File: packages/core/src/browser/preferences/preference-service.ts [hunk 4]
```diff
@@ -27,11 +27,30 @@ import { PreferenceConfigurations } from './preference-configurations';
...
+/**
+ * Representation of a preference change.
+ */
 export interface PreferenceChange {
+    /**
+     * The name of the changed preference.
+     */
     readonly preferenceName: string;
+    /**
+     * The new value of the changed preference.
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-service.ts:39]: what is the meaning if this is not set? Same for oldvalue

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 5]
```diff
@@ -56,7 +81,10 @@ const OVERRIDE_PROPERTY = '\\[(.*)\\]$';
 export const OVERRIDE_PROPERTY_PATTERN = new RegExp(OVERRIDE_PROPERTY);
 
 const OVERRIDE_PATTERN_WITH_SUBSTITUTION = '\\[(${0})\\]$';
-
+/**
+ * Specialized {@link FrontendApplicationConfig} that can be used to configure default
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:85]:
```suggestion
 * Specialized {@link FrontendApplicationConfig} to configure default
```

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 6]
```diff
@@ -56,7 +81,10 @@ const OVERRIDE_PROPERTY = '\\[(.*)\\]$';
...
+ * preference values for the {@link PreferenceSchemaProvider}.
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:86]: Are these default values really only used by the PreferenceSchemaProvider?

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 7]
```diff
@@ -68,6 +96,11 @@ export namespace FrontendApplicationPreferenceConfig {
     }
 }
 
+/**
+ * The {@link PreferenceSchemaProvider} collects all {@link PreferenceContribution}s and combines
+ * the preference schema provided by these contributions into one collective schema. The preferences which
+ * are managed by this {@link PreferenceProvider} are dervied from this combined schema.
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:102]:
```suggestion
 * are provided by this {@link PreferenceProvider} are dervied from this combined schema.
```

File: packages/core/src/browser/preferences/preference-contribution.ts [hunk 8]
```diff
@@ -96,6 +129,15 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
     }
 
     protected readonly overrideIdentifiers = new Set<string>();
+
+    /**
+     * Register a new overrideIdentifier. Existing indentifiers are not replaced.
+     *
+     * Useful to potentially override existing values while keeping both values in store
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-contribution.ts:136]:
```suggestion
     * Allows overriding existing values while keeping both values in store
```

File: packages/core/src/browser/preferences/preference-provider.ts [hunk 9]
```diff
@@ -48,7 +48,11 @@ export interface PreferenceResolveResult<T> {
     configUri?: URI
     value?: T
 }
-
+/**
+ * The {@link PreferenceProvider} is used to store and retrieve preference values. A {@link PreferenceProvider} does not operate in a global scope and is typically
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-provider.ts:52]:
```suggestion
 * The {@link PreferenceProvider} is used to store and retrieve preference values. A {@link PreferenceProvider} does not operate in a global scope but is
```

File: packages/core/src/browser/preferences/preference-provider.ts [hunk 10]
```diff
@@ -118,10 +122,29 @@ export abstract class PreferenceProvider implements Disposable {
         return false;
     }, 0);
 
+    /**
+     * Retrieve the stored value for the given preference and resource URI.
+     *
+     * @param preferenceName the preference identifier.
+     * @param resourceUri the uri of the resource for which the preference is stored. This used to retrieve
```
Comment (JonasHelming) [inline packages/core/src/browser/preferences/preference-provider.ts:129]:
```suggestion
     * @param resourceUri the uri of the resource for which the preference is stored. This is used to retrieve
```

File: packages/core/src/browser/preferences/preference-contribution.ts (hunk 6 above)
Comment (tortmayr) [reply packages/core/src/browser/preferences/preference-contribution.ts:86]: Yes. At the moment they are only used in the PreferenceSchemaProvider

File: packages/core/src/browser/preferences/preference-service.ts (hunk 4 above)
Comment (tortmayr) [reply packages/core/src/browser/preferences/preference-service.ts:39]: Added a short explanation

**Comment by sdirix** [comment]
Merged upstream with https://github.com/eclipse-theia/theia/pull/8677

**Pull request closed without merging**